### DISCUSSION TITLE
Unnatural Chat Interaction Behavior

### DISCUSSION THREAD

**Discussion by ProjectMoon** [Taewan-P/gpt_mobile#87] [General]
Love the app.  Been looking for something that works with ollama on mobile aside from the Open WebUI PWA.

One thing that seems odd to me: it seems like user messages are being joined together and sent as one huge message every time to the ollama endpoint. When the LLM responds, it responds as if all of my previous messages are one. First it'll talk about topic 1 (from message 1), then topic 2 from message 2, and so on, until it finally answers the most recent message.

Is it something to do with how the app is assembling message history?

**Comment by Taewan-P** [comment]
Hello!
Thanks for the feedback!
After further investigation, I found out that there is a bug when assembling Ollama message history. I will convert this to a issue since this behavior is not intended.
